netman/core/switch_sessions.py

```diff
@@ -49,11 +52,31 @@ def open_session(self, switch, session_id):
             raise
 
         self.logger.info("Switch for session {} connected and in transaction mode, storing session".format(session_id))
-        self.sessions[session_id] = switch
+        self.add_session(switch, session_id)
         self._start_timer(session_id)
 
         return session_id
 
+    def add_session(self, switch, session_id):
+        if session_id in self.sessions:
```

Comment:
This check is already done in open_session if you get to this point the session doesn't exist